libs/cardano-ledger-binary/cardano-ledger-binary.cabal

```diff
@@ -1,6 +1,6 @@
 cabal-version: 3.0
 name:          cardano-ledger-binary
-version:       1.4.0.0
+version:       1.4.1.0
```

Comment:
I think this requires `1.5.0.0`, because it's a breaking change .

Reply:
The change won't break any client usage of the library, because it just introduces a new constructor. So a `minor` bump should be enough I thought.

Reply:
```suggestion
version:       1.5.0.0
```

@teodanciu is right! This is a major breaking change. This is not explicitly listed in PVP, but it is widely considered to be a breaking change in the Haskell community. See some discussions [here](https://discourse.haskell.org/t/does-adding-a-data-constructor-require-a-major-version-bump-according-to-the-pvp/3600) and [comment by Phadej: "It adds constructors to LicenseId and LicenseListVersion, Language.Haskell.Extension.Extension. These are breaking changes."](https://github.com/haskell/cabal/issues/9258#issue-1895931692)

Here is my reason depicted with an example why I consider such change to be a breaking change:

Imagine package `A-1.0.0` defines `Foo`:

```haskell
data Foo = Foo | Bar deriving Read
```

Package `B-1.0.0` depends on `A >= 1.0` and provides a function:

```haskell
readFooIxMaybe :: String -> Maybe Int
readFooIxMaybe str = do
  foo <- readMaybe str
  pure $ case foo of
    Foo -> 1
    Bar -> 2
```

If you add a constructor to `Foo` and only bump a minor version in package `A-1.0.1`:

```haskell
data Foo = Foo | Bar | Baz deriving Read
```

Then package `B-1.0.0` will have the `readFooIxMaybe` function turn from a total to a partial function.

Reply:
Got it! How unthoughtful of me. Bumping!